include/fsm/fill_spill_merge.hpp

```diff
@@ -539,18 +546,19 @@ static dh_label_t OverflowInto(
   //SECOND PLACE TO STASH WATER: IN THIS DEPRESSION'S NEIGHBOUR
   //Maybe we can fit it into this depression's overflow depression!
 
+
   auto &pdep = deps.at(this_dep.parent);
   if(this_dep.odep==NO_VALUE){      //Does the depression even have such a neighbour? 
-    if(this_dep.parent!=OCEAN && pdep.water_vol==0) //At this point we're full and heading to our parent, so it needs to know that it contains our water
-      pdep.water_vol += this_dep.water_vol;
+    if(this_dep.parent!=OCEAN && pdep.water_vol==0 && (pdep.lchild == this_dep.dep_label || pdep.rchild == this_dep.dep_label)) //At this point we're full and heading to our parent, so it needs to know that it contains our water
+      pdep.water_vol += deps.at(pdep.lchild).dep_vol + deps.at(pdep.rchild).dep_vol;//this_dep.water_vol;
```

Comment:
Why? Also: it would be good to remove the (`//this_dep.water_vol`) comment from the end of the line. The old code appears in the diff and it's better not to live litter around in the codebase.

Reply:
Same reasoning as above and also, I changed += water_vol to dep_vol since I wasn't sure if the current water_vol may sometimes be larger than the current dep_vol as it is still busy overflowing (on reflection, I don't think that should happen with the way the code is structured, but I am not certain). The pdep will still receive any additional water as extra_water so I was concerned about the extra_water being added twice, essentially.
I am not sure why we would have added only one child's volume to its parent, ever. It seems like this would have permanently neglected adding the volume of the second child.

Reply:
You're absolutely right on the commented piece!